tests/cases/compiler/parseNoWhitespaceBeforeExclamation.ts

```diff
@@ -0,0 +1,5 @@
+declare var value: any;
+
+value !;
+value !instanceof String;
+value/* this is a comment */! instanceof String;

```

Comment:
```suggestion
value !;

value !instanceof String;
value ! instanceof String
value/* this is a comment */!instanceof String;
value/* this is a comment */! instanceof String;

value !in String;
value ! in String
value/* this is a comment */!in String;
value/* this is a comment */! in String;
```

Reply:
Sure though I'll note that none of the code here actually depends on what happens after 😄